new_course/1_programming_fundementals/methods.md

```diff
@@ -176,29 +176,62 @@ end
 puts my_name #=> "Joe Smith"
 ~~~
 
+This can be useful for example in a situation where you want your method to check for possible errors before continuing:
+
+~~~ruby
+def even_odd(number)
+  unless number.is_a? Numeric
+    return "A number was not entered"
+  end
+
+  if number % 2 == 0
+    "That is an even number."
+  else
+    "That is an odd number."
+  end
+end
+
+puts even_odd(20) #=>  That is an even number.
+puts even_odd("Ruby") #=>  A number was not entered.
+~~~
+
+Try removing the explicit `return` from the method above. Did the method run as expected?
+
 #### Difference Between `puts` and `return`
 
 A common source of confusion for new programmers is the difference between `puts` and `return`.
 
- * `puts` is a method that prints whatever argument you pass it to the console.
- * `return` is the final output of a method that you can use in other places throughout your code. 
+ * `puts` is a method that simply prints a given value to the console.
+ * `return` produces a value that can be used in your code. 
  
-For example, you can create a method that calculates the product of 3 * 10:
+For example:
 
 ~~~ruby
-def dirty_thirty
-  3 * 10
+def squared(number)
+  puts number * number
 end
 ~~~
 
-You can then interact with this outcome in other places in your code:
+The above method will only print to the console the value of a given number squared. However, it will not produce or return that value.  So writing something like `x = squared(20)` will print `400` to the console but the variable `x` will not be assigned the value `400` but rather it will have a `nil` value.  This is because the method did not actually produce a value.
+
+If we instead write the same method using a implicit or explicit return:
```

Comment:
```suggestion
Now, let's write the same method but with an implicit return instead of `puts`. (Using an explicit return will act exactly the same in this example.)
```